sdk/core/core-lro/docs/LROEngine.md

```diff
@@ -0,0 +1,214 @@
+
+### [`coreClientLro`](https://github.com/deyaaeldeen/autorest.typescript/blob/lro-abstract/src/coreClientLro.ts) and [`coreHttpLro`](https://github.com/deyaaeldeen/autorest.typescript/blob/lro-abstract/src/coreHttpLro.ts)
+
+I propose to make these classes auto-generated by Autorest. These classes implement `LongRunningOperation<T>` and various bits of `coreHttpLro` have been showed earlier. These classes will need access to a few pieces: operation specification and operation arguments and a primitive function that can take them as input to send a request and also converts the response into one of type `LroResponse<T>` which has both the flattened and the raw responses. Furthermore, those classes can also, optionally, take as input an object of type `LroResourceLocationConfig`, which could determine where to find the results of the LRO after the operation finished. Typically, Autorest figures out the value for `LroResourceLocationConfig` from the `x-ms-long-running-operation-options` swagger extension.
+
+## Usage examples
+
+### Create an object of `coreClientLro`
+
+```ts
+const directSendOperation = async (
+  args: coreClient.OperationArguments,
+  spec: coreClient.OperationSpec
+): Promise<unknown> => {
+  return this.client.sendOperationRequest(args, spec);
+};
+const sendOperation = async (
+  args: coreClient.OperationArguments,
+  spec: coreClient.OperationSpec
+) => {
+  let currentRawResponse: coreClient.FullOperationResponse | undefined = undefined;
+  const providedCallback = args.options?.onResponse;
+  const callback: coreClient.RawResponseCallback = (
+    rawResponse: coreClient.FullOperationResponse,
+    flatResponse: unknown
+  ) => {
+    currentRawResponse = rawResponse;
+    providedCallback?.(rawResponse, flatResponse);
+  };
+  const updatedArgs = {
+    ...args,
+    options: {
+      ...args.options,
+      onResponse: callback
+    }
+  };
+  const flatResponse = await directSendOperation(updatedArgs, spec);
+  return {
+    flatResponse,
+    rawResponse: {
+      statusCode: currentRawResponse!.status,
+      body: currentRawResponse!.parsedBody,
+      headers: currentRawResponse!.headers.toJSON()
+    }
+  };
+};
+
+const lro = new CoreClientLro(
+  sendOperation,
+  { options }, // arguments are just the operation options
+  spec
+);
+```
+
+### Using `LroEngine`
+
+```ts
+const pollerEngine = new LroEngine(lro, { intervalInMs: 2000 }); // lro was instantiated in the previous section
+const result = pollerEngine.pollUntilDone();
+```
+
```

Comment:
Wouldn’t it be possible to create LROs specific to the known Azure specs with the core-lro classes that are already available?